rotkehlchen/chain/evm/decoding/spark/savings/decoder.py

```diff
@@ -0,0 +1,321 @@
+                to_address=MIGRATION_ACTIONS_CONTRACT,
+            ))
+
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.DEPOSIT and
+                    event.event_subtype == HistoryEventSubType.DEPOSIT_ASSET and
+                    event.asset == A_SDAI and
+                    event.location_label == received_address
+            ):  # sDAI -> sUSDS
+                event.event_type = HistoryEventType.MIGRATE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.address = MIGRATION_ACTIONS_CONTRACT
+                for tx_log in context.all_logs:  # find the sDAI withdraw/redeem log and get the assets (DAI) amount  # noqa: E501
+                    if tx_log.topics[0] == SDAI_REDEEM and bytes_to_address(tx_log.topics[1]) == MIGRATION_ACTIONS_CONTRACT and bytes_to_address(tx_log.topics[3]) == received_address:  # noqa: E501
+                        event.notes = f'Migrate {event.amount} sDAI ({(underlying_amount := token_normalized_value_decimals(token_amount=int.from_bytes(context.tx_log.data[:32]), token_decimals=DEFAULT_TOKEN_DECIMALS))} DAI) to sUSDS'  # noqa: E501
+                        event.extra_data = {'underlying_amount': str(underlying_amount)}
+                        event.counterparty = CPT_SKY
+                        break  # found the log
+                else:
+                    log.error(f'Could not find the log of the sDAI Withdraw in {context.transaction}')  # noqa: E501
+                    event.notes = f'Migrate {event.amount} sDAI to sUSDS'
+
+                break  # found the event
+
+            elif (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.asset == A_DAI and
+                    event.location_label == received_address
+            ):  # DAI -> sUSDS
+                action_items[0].to_notes = f'Receive {shares_amount} sUSDS ({assets_amount} USDS) from DAI to sUSDS migration'  # noqa: E501  # fix the notes of the receival to be DAI
+                event.event_type = HistoryEventType.MIGRATE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.address = MIGRATION_ACTIONS_CONTRACT
+                event.notes = f'Migrate {event.amount} DAI to sUSDS'
+                event.counterparty = CPT_SKY
+                break  # found the event
+
+        else:
+            log.error(f'Could not find the deposit sDAI event in {context.transaction}')
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(action_items=action_items)
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        decoders = dict.fromkeys(self.spark_savings_tokens, (self._decode_spark_tokens_deposit_withdrawal,))  # noqa: E501
+        if self.psm_address is not None:
```

Comment:
this is only available in ethereum? If so you should extend the address_to_decoders in the ethereum subclass

Reply:
> this is only available in ethereum? If so you should extend the address_to_decoders in the ethereum subclass

it isn't available in ethereum and gnosis.